### ISSUE TITLE
Poor perfomance in Windows

### ISSUE THREAD

**Issue by matiaslopezd** [google/mediapipe#4214]
### Have I written custom code (as opposed to using a stock example script provided in MediaPipe)

Yes

### OS Platform and Distribution

Windows 10 x64

### MediaPipe version

0.1.1675465747

### Bazel version

-

### Solution

SelfieSegmentation

### Programming Language and version

Javascript

### Describe the actual behavior

```shell
We achieve a rendering process practically "free" with Dedicated Worker techniques. But we test Selfie Segmentation standalone without the rendering process (canvas). In Linux and MacOS the CPU run between 6% and 15%.

In Windows, the average consumption can go from ~30% to 100%. We detect if GPU is present to set the `useCpuInference` to true or false before instantiating, but in devices with GPU, the consumption is still higher than in Linux and MacOS.

With 100% of the CPU, the apps have weird behavior that blocks the main thread also network tasks, and more.
```

### Describe the expected behaviour

```shell
We expect similar CPU performance that Linux and MacOS, maybe a range of 10%-30% more but 100% means can't run anything except segmentation.

There is a way to contribute with a non-minified source code? I read a new version is in progress, will be able to contribute to the source code without minification?
```

### Standalone code/steps you may have used to try to get what you need

```shell
Check this [codesandbox](https://681o5y.csb.app/), try it on Linux and Windows, and check in Linux easily run under ~30%, and in Windows goes to 100% CPU.

The sandbox only uses Selfie segmentation without a canvas rendering process to not add CPU workload to the benchmark test.

Source code of codesandbox:


<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
    <title>Selfie segmentation performance test</title>
    <style>
      video {
        width: 640px;
        height: 480px;
      }
    </style>
  </head>
  <body>
    <h1>Selfie segmentation performance test</h1>
    <video id="original" muted autoplay playsinline></video>
    <script
      src="https://cdn.jsdelivr.net/npm/@mediapipe/selfie_segmentation/selfie_segmentation.js"
      crossorigin="anonymous"
    ></script>
    <!--<script src="virtual-background.js"></script>-->
    <script>
      const video = document.querySelector("#original");
      const segmentation = new SelfieSegmentation({
        locateFile: (file) =>
          `https://cdn.jsdelivr.net/npm/@mediapipe/selfie_segmentation/${file}`
      });
      segmentation.setOptions({ modelSelection: 1, useCpuInference: false });
      segmentation.onResults((frame) => {
        // Discard frames to avoid increased CPU usage with non-segmentation task
      });
      async function callback() {
        await segmentation.send({ image: video });
        video.requestVideoFrameCallback(callback);
      }

      navigator.mediaDevices
        .getUserMedia({ video: true })
        .then(async (camera) => {
          video.srcObject = camera;
          video.requestVideoFrameCallback(callback);
        });
    </script>
  </body>
</html>
```

### Other info / Complete Logs

```shell
All CPU measurement comes from DevTools Performance Monitor, not a native monitor or task manager.
```

**Comment by kuaashish** [comment]
Hello @matiaslopezd,
Would you please validate which mode of TFLite InferewnceCalculator is actually running on windows? WebGPU support is very recent and maybe not enabled in Chrome on windows browser. Further, We are reassigning issue to right owner to understand more about the WebGPU support. Thank you

**Comment by danrossi** [comment]
I've not experienced a problem apart from high cpu usage of canvas capture stream at 1080p due to not being hardware accelerated. Its 7-10% cpu and doing the mix in wegl 8% gpu. But WebGPU would be nice. And doing the actual mask mix with background and video, and do a blur filter on the mask in wasm would be good. There is a calculator for that already not enabled and built in to selfiesegmentation. 

The examples make people do canvas software rendering of video frames afterwards. Tensorflow was also grabbing video frames via software also so had to use webgl as the buffer input before. I see mediapipe use webgl to grab the video texture. 

I am returning the mask and video texture to mix them in a WebGL shader but may have to update it to WebGPU support. I am also experiencing high cpu usage for video textures with WebGPU in chrome and trying to figure out how to report it. So that is a problem. 

example fragment mix after a blur filter on the mask edges ```outColor = mix(bgTex, vec4(frameColor, 1.0), maskVal);```

**Comment by kuaashish** [comment]
@matiaslopezd,

You are currently utilising the legacy MediaPipe Selfie Segmentation solution, which is no longer maintained by our team, We kindly advise you to refer to our [guide](https://developers.google.com/mediapipe/solutions/guide#legacy), where it is indicated that the Selfie Segmentation has been upgraded to the new Image Segmentation.

We earnestly request that you transition to our new Image Segmentation Task API as outlined in the provided documentation [here](https://developers.google.com/mediapipe/solutions/vision/image_segmenter/web_js). Additionally, you can find a [steps to setup Image Segmentation in Web](https://developers.google.com/mediapipe/solutions/vision/image_segmenter/web_js) in the same resource for your reference. Also, we request please report back if you observe similar behaviour in the upgraded task API. Thank you

**Comment by danrossi** [comment]
The edges of the deeplabv3 is inaccurate and picks up other objects in the background so the selfie segment model is still required to be used in the new api. The alpha channel is inverted in deeplabv3 also so the r channel used has to be inverted when doing a mix in webgl. deeplabv3 has enabled softmax smoothing where the other model doesn't.
